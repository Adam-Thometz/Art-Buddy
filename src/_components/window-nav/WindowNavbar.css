.WindowNavbar-main {
  position: relative;
  display: flex;
  justify-content: space-around;
  width: 50%;
  margin: 0 auto;
}

.WindowNavbar-games {
  display: flex;
  margin: 0 auto;
}

.WindowNavbar-button {
  font-size: 3rem;
  cursor: pointer;
  padding-bottom: 5px;
}

.WindowNavbar-label {
  font-size: 3rem;
  margin: 0 auto;
}

.underlined {
  background-image: url(_icons/underline.png);
  background-repeat: no-repeat;
  background-position: bottom;
  background-size: 100%;
}