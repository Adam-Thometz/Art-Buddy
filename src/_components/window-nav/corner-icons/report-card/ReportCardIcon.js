import React from 'react';
import { useParams } from 'react-router-dom';

import Icon from '../../../icon/Icon';
import Popup from '../../../popup/Popup';

import { reportCardIcon } from '../../../../instrument-id/_icons/iconImports';
import ReportCard from '../../../../instrument-id/play/report-card/ReportCard';

const ReportCardIcon = () => {
  const { level } = useParams()
  const popupTrigger = (<Icon icon={reportCardIcon} text='REPORT CARD' size='50px' />)
  return (
    <Popup
      title='REPORT CARD'
      trigger={popupTrigger}
      triggerClass='WindowNavbar-corner-icon report-card'
      popup={<ReportCard level={level} />}
    />
  )
}

export default ReportCardIcon;