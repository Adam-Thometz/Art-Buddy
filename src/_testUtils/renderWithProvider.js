import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;