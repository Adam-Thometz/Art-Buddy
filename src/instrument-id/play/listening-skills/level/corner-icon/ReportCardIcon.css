.ReportCardIcon {
  position: absolute;
  right: 3%;
  top: 75px;
}