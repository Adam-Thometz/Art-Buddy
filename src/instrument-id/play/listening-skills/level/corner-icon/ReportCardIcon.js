import React, { useContext } from 'react';
import { useParams } from 'react-router-dom';
import { PopupContext } from '_context/PopupContext';

import Icon from '_components/icon/Icon';
import ReportCard from '_components/report-card/ReportCard';

import { reportCardIcon } from '_media/instrument-id/_icons/iconImports';

const ReportCardIcon = () => {
  const { level } = useParams();
  const { setCurrPopup } = useContext(PopupContext)
  
  const openReportCard = () => setCurrPopup({
    title: 'REPORT CARD',
    popup: <ReportCard game="instrumentId" level={level} />
  });

  return <Icon icon={reportCardIcon} text='REPORT CARD' size='50px' width='34px' onClick={openReportCard} />;
};

export default ReportCardIcon;