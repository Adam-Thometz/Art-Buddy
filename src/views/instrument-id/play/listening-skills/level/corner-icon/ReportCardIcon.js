import { useContext, useEffect } from "react";
import { useParams } from "react-router-dom";
import { PopupContext } from "context/PopupContext";

import Icon from "components/icon/Icon";
import ReportCard from "components/report-card/ReportCard";

import { REPORT_CARD } from "assets/_general/general.assets";

const ReportCardIcon = () => {
  const { level } = useParams();
  const { setCurrPopup } = useContext(PopupContext);

  useEffect(() => {
    return () => setCurrPopup(null);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const openReportCard = () =>
    setCurrPopup({
      title: "REPORT CARD",
      popup: <ReportCard game="instrumentId" level={level} />,
    });

  return (
    <Icon
      icon={REPORT_CARD}
      text="REPORT CARD"
      size="50px"
      width="34px"
      onClick={openReportCard}
    />
  );
};

export default ReportCardIcon;
