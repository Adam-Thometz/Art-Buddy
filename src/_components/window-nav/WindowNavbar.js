import React from "react";
import { useLocation } from "react-router-dom";

import { useSelector, useDispatch } from "react-redux";
import { changeMenuGames } from "_redux/actions/mainSettingsActions";

import './WindowNavbar.css';

import BackArrow from "../back-arrow/BackArrow";

const WindowNavbar = ({ cornerIcon = null, page = null }) => {
  const dispatch = useDispatch();
  const location = useLocation();

  const menuCategory = useSelector(state => state.mainSettings.menu);

  const handleChange = label => dispatch(changeMenuGames(label));
  const underline = target => menuCategory === target ? ' underlined' : '';

  const menuPage = () => (
    <header className="WindowNavbar-main">
      <span onClick={() => handleChange('games')} className={`WindowNavbar-button${underline('games')}`}>GAMES</span>
      <span onClick={() => handleChange('tools')} className={`WindowNavbar-button${underline('tools')}`}>TOOLS</span>
    </header>
  );

  const regularPage = () => (
    <header className="WindowNavbar-games">
      {location.pathname.length === 1 ? null : <BackArrow />}
      <h1 className="WindowNavbar-label">{page}</h1>
      {cornerIcon ? (
        <div className="WindowNavbar-corner">{cornerIcon}</div>
      ) : null}
    </header>
  );

  return page ? regularPage() : menuPage();
};

export default WindowNavbar;