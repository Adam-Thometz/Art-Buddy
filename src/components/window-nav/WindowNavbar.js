import { useContext, useEffect } from "react";
import { useLocation } from "react-router-dom";
import { PopupContext } from "context/PopupContext";

import { useSelector, useDispatch } from "react-redux";
import { changeMenu } from "store/_general/generalReducer";

import "./WindowNavbar.css";

import BackArrow from "./back-arrow/BackArrow";
import Help from "components/help/Help";

const WindowNavbar = ({ cornerIcon = null, page = null }) => {
  const { menu } = useSelector((state) => state.general);
  const { setCurrPopup } = useContext(PopupContext);
  const dispatch = useDispatch();
  const location = useLocation();

  useEffect(() => {
    return () => setCurrPopup(null);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [page]);

  const handleChange = (e) => dispatch(changeMenu(e.target.id));
  const underline = (target) => (menu === target ? " underlined" : "");

  const menuPage = (
    <header className="WindowNavbar-main">
      <span
        onClick={handleChange}
        id="games"
        className={`WindowNavbar-button${underline("games")}`}
      >
        GAMES
      </span>
      <span
        onClick={handleChange}
        id="tools"
        className={`WindowNavbar-button${underline("tools")}`}
      >
        TOOLS
      </span>
    </header>
  );

  const regularPage = (
    <header className="WindowNavbar-games">
      {location.pathname.length === 1 ? null : <BackArrow />}
      <h1 className="WindowNavbar-label">{page}</h1>
      {cornerIcon ? (
        <div className="WindowNavbar-corner">{cornerIcon}</div>
      ) : null}
      {location.pathname.length === 1 ? null : <Help />}
    </header>
  );

  return page ? regularPage : menuPage;
};

export default WindowNavbar;
