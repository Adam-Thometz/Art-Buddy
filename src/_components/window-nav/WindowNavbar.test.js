import React from "react";
import renderWithProvider from '../../_testUtils/renderWithProvider';
import { screen } from "@testing-library/react";
import WindowNavbar from './WindowNavbar'
import ReportCardIcon from "./corner-icons/report-card/ReportCardIcon";

describe('WindowNavbar component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<WindowNavbar />);
  });

  it('matches the snapshot of the WindowNavbar', () => {
    const { asFragment } = renderWithProvider(<WindowNavbar />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it('renders the WindowNavbar in the music menu', () => {
    renderWithProvider(<WindowNavbar page='MUSIC GAMES' />);
    expect(screen.getByText('MUSIC GAMES')).toBeInTheDocument();
  });

  it('renders a corner icon', () => {
    renderWithProvider(<WindowNavbar page='Listening Skills Test' cornerIcon={<ReportCardIcon />}/>);
    expect(screen.getByText('Listening Skills Test')).toBeInTheDocument();
    expect(screen.getByText('REPORT CARD')).toBeInTheDocument();
  });
})