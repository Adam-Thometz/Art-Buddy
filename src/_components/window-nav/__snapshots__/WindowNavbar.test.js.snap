// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindowNavbar component matches the snapshot of the WindowNavbar 1`] = `
<DocumentFragment>
  <header
    class="WindowNavbar-main"
  >
    <span
      class="WindowNavbar-button underlined"
      id="games"
    >
      GAMES
    </span>
    <span
      class="WindowNavbar-button"
      id="tools"
    >
      TOOLS
    </span>
  </header>
</DocumentFragment>
`;
